crates/biome_json_syntax/src/file_source.rs

```diff
@@ -11,6 +12,30 @@ use core::str;
 pub struct JsonFileSource {
     allow_trailing_commas: bool,
     allow_comments: bool,
+    variant: JsonFileVariant,
+}
+
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[derive(
+    Debug, Clone, Default, Copy, Eq, PartialEq, Hash, serde::Serialize, serde::Deserialize,
+)]
+#[serde(rename_all = "camelCase")]
+/// It represents the extension of the file
```

Comment:
I think we should doc strings before the annotations. It's what we do in most places already, and is also the convention in the Rust standard library.